### PR TITLE
False positive matrix

### PR DESCRIPTION
**Pull request by GFabien**
Add a new metric to compute matrix of false positives between classes. Expect `binarize_target` to be `True` in `MetricOverlay`.

### CODE REVIEW

File: segmentation/metrics/overlap_metrics.py [hunk 1]
```diff
@@ -51,3 +52,20 @@ def mean_false_negatives(self, prediction, target):
...
+
+    @staticmethod
+    def false_positive_matrix(prediction, target):
+        """
+        Compute matrix of false positives between channels.
+        Target is expected to be a discrete one hot encoded label map.
+        Prediction can be discrete or continuous.
+        """
+        n_channels = target.shape[1]
+        fp_matrix = torch.zeros((n_channels, n_channels)).to(prediction.device)
```
Comment (romainVala) [inline segmentation/metrics/overlap_metrics.py:64]: c'est un détail ici, car la matrice est petite, donc peu de donnée a transferer, mais autant prendre de bonne habitude,
J'ai lu quelque part que l'utilisation de .to(device) est a éviter car elle implique un transfert des données cpu to gpu.
Du coup quand c'est possible il faut initialise directement sur le gpu
https://discuss.pytorch.org/t/creating-tensors-on-gpu-directly/2714
je sais pas si il y a d'autre endroit dans le code, ou on pourait corriger, 
d=time.time(); torch.cuda.FloatTensor(1000, 1000).fill_(0)  ; print('time {}'.format(time.time()-d));                                                                                              
time 0.004454374313354492

 d=time.time(); torch.zeros(1000, 1000, device=torch.device('cuda'))  ; print('time {}'.format(time.time()-d));                                                                                     
time 0.004109859466552734
d=time.time();  torch.zeros(1000, 1000).to(device=torch.device('cuda'))  ; print('time {}'.format(time.time()-d));                                                                                                          
time 0.02329540252685547

d'un point de vue syntaxe je prefère
 torch.zeros(1000, 1000, device=torch.device('cuda'))

File: segmentation/run_model.py [hunk 2]
```diff
@@ -582,12 +584,12 @@ def record_simple(self, df, sample, predictions, targets,
                 for metric in self.metrics:
                     name = f'metric_{metric["name"]}'
 
-                    info[name] = to_numpy(
+                    info[name] = json.dumps(to_numpy(
```
Comment (romainVala) [inline segmentation/run_model.py:587]: juste pour comprendre, pourquoi ce changement ? tu aurais pu garder un vecteur de float dans info plutot que des strings ?

Comment (GFabien) [reply segmentation/run_model.py:587]: J'aurais pu mais le problème c'est qu'après sauvegarde en CSV, la matrice devient très dur à parser dans l'autre sens. Ici il suffit de faire quelque chose du genre :
```python
import ast
x = df['metric_fp_matrix'][0]
array = np.array(ast.literal_eval(x))
```

Comment (GFabien) [reply segmentation/run_model.py:587]: Oui en fait ça marche aussi, j'avais voulu essayer ça mais je m'étais trompé, j'avais fait json.dumps au lieu de json.loads.
